### PR TITLE
Property 'mutedSegmentData' must allow null

### PR DESCRIPTION
**Pull request by bishil06**
Type:  #Bugfix

Fixes: #388 

`mutedSegmentData`
Array of muted segments in the video.
If there are no muted segments, the value will be `null`.
[twitch reference](https://dev.twitch.tv/docs/api/reference#get-videos)

my solution 
```ts
muted_segments: HelixVideoMutedSegmentData[];  // -> muted_segments: HelixVideoMutedSegmentData[] | null; 
```

### CODE REVIEW

File: packages/api/src/api/helix/video/HelixVideo.ts [hunk 1]
```diff
@@ -215,8 +215,12 @@ export class HelixVideo extends DataObject<HelixVideoData> {
 	/**
 	 * The raw data of muted segments of the video.
 	 */
-	get mutedSegmentData(): HelixVideoMutedSegmentData[] {
-		return this[rawDataSymbol].muted_segments.slice();
+	get mutedSegmentData(): HelixVideoMutedSegmentData[] | null {
```
Comment (d-fischer) [inline packages/api/src/api/helix/video/HelixVideo.ts:218]: This is an unnecessary breaking change. I'd rather return an empty array if Twitch returns null.

File: packages/api/src/api/helix/video/HelixVideo.ts [hunk 2]
```diff
@@ -215,8 +215,12 @@ export class HelixVideo extends DataObject<HelixVideoData> {
...
+		if (this[rawDataSymbol].muted_segments === null) {
```
Comment (d-fischer) [inline packages/api/src/api/helix/video/HelixVideo.ts:219]: This is a prime example for usage of null coalescing / optional chaining.

**Review by d-fischer — CHANGES_REQUESTED** [review]
Apart from the proposed direct changes, this PR should probably be on the `versions/5.2` branch.

**Comment by bishil06** [comment]
Modified to return `[]` using the nullish coalescing operator if `mutedSegmentData` is `null`

https://github.com/twurple/twurple/blob/058d669ed0bc24d05b833a8bc01963ab4f10f97d/packages/api/src/api/helix/video/HelixVideo.ts#L218-L220

**Comment by d-fischer** [comment]
Why did you undo the change to the internal data interface? It should be true to the original Twitch data as exactly as possible. If Twitch can return null, it should include null. This does *not* affect the getter, and indeed resolves the linter issue shown in the build log.

**Comment by bishil06** [comment]
It was my mistake. I'm sorry

File: packages/api/src/api/helix/video/HelixVideo.ts [hunk 3]
```diff
@@ -216,7 +216,7 @@ export class HelixVideo extends DataObject<HelixVideoData> {
 	 * The raw data of muted segments of the video.
 	 */
 	get mutedSegmentData(): HelixVideoMutedSegmentData[] {
-		return this[rawDataSymbol].muted_segments.slice();
+		return this[rawDataSymbol].muted_segments === null ? [] : this[rawDataSymbol].muted_segments;
```
Comment (d-fischer) [inline packages/api/src/api/helix/video/HelixVideo.ts:219]: Again, optional chaining here.

**Review by d-fischer — CHANGES_REQUESTED** [review]
Also still not on the version branch.

**Pull request closed without merging**